meld/cluster_topdown.py

```diff
@@ -0,0 +1,305 @@
+import numpy as np
+import heapq
+from .cluster import sparse_dim_connectivity
+from numba import float64, jitclass, jit
+
+
+@jit(nopython=True)
+def tfce_calc(extent, param_e, upper_limit, h_plus1, lower_limit):
+
+    return np.power(extent, param_e) * \
+                (np.power(upper_limit, h_plus1)
+                 - np.power(lower_limit, h_plus1)) / h_plus1
+
+clus_spec = [
+    ()
+]
+
+
+class Cluster():
+    """ TFCE cluster class
+    """
+    def __init__(self, param_e, param_h):
+        """
+            Initialize empty cluster
+        """
+        self.param_e = param_e
+        self.h_plus1 = param_h + 1
```

Comment:
These parameters will never be different per-cluster.  They are meant to be private implementation helpers only, and thus can be "messy" and not self-contained.  I don't know if it helps python speed to have them this way.